From 78230f6cdb041fcc4041fc64f57108a812782ab1 Mon Sep 17 00:00:00 2001
From: Gilly1970 <gilroyscott@hotmail.com>
Date: Sat, 20 Sep 2025 17:58:37 +1000
Subject: [PATCH] 9997-kernel-6.12-EEPROM-contains-0s

---
 mt7996/eeprom.c | 50 ++++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 49 insertions(+), 1 deletion(-)

diff --git a/mt7996/eeprom.c b/mt7996/eeprom.c
index da3231c..df0b6d2 100644
--- a/mt7996/eeprom.c
+++ b/mt7996/eeprom.c
@@ -93,6 +93,44 @@ mt7996_eeprom_parse_stream(const u8 *eeprom, u8 band_idx, u8 *path,
 	}
 }
 
+static void
+mt7996_eeprom_fixup_tx_power(struct mt7996_dev *dev, const u8 *def)
+{
+	u8 *eeprom = dev->mt76.eeprom.data;
+	int i;
+	bool zeros_detected = false;
+	bool sixg_overridden = false;
+
+	if (!eeprom[MT_EE_TX0_POWER_2G]) {
+		eeprom[MT_EE_TX0_POWER_2G] = def[MT_EE_TX0_POWER_2G];
+		zeros_detected = true;
+	}
+
+	for (i = MT_EE_TX0_POWER_5G; i < MT_EE_TX0_POWER_5G + 5; ++i) {
+		if (!eeprom[i]) {
+			eeprom[i] = def[i];
+			zeros_detected = true;
+		}
+	}
+
+	/*
+	 * Unconditionally use default firmware values for the 6GHz band.
+	 * This is a workaround for hardware with incorrect (but non-zero)
+	 * tx_power values in the EEPROM.
+	 */
+	for (i = MT_EE_TX0_POWER_6G; i < MT_EE_TX0_POWER_6G + 8; ++i) {
+		if (eeprom[i] != def[i]) {
+			eeprom[i] = def[i];
+			sixg_overridden = true;
+		}
+	}
+
+	if (zeros_detected)
+		dev_warn(dev->mt76.dev, "eeprom tx_power zeros detected, using defaults\n");
+	if (sixg_overridden)
+		dev_warn(dev->mt76.dev, "Overriding 6GHz eeprom tx_power with firmware defaults\n");
+}
+
 static bool mt7996_eeprom_variant_valid(struct mt7996_dev *dev, const u8 *def)
 {
 #define FEM_INT	0
@@ -148,6 +186,8 @@ mt7996_eeprom_check_or_use_default(struct mt7996_dev *dev, bool use_default)
 		goto out;
 	}
 
+	mt7996_eeprom_fixup_tx_power(dev, fw->data);
+
 	if (!use_default && mt7996_eeprom_variant_valid(dev, fw->data))
 		goto out;
 
@@ -212,7 +252,15 @@ static int mt7996_eeprom_load(struct mt7996_dev *dev)
 				goto out;
 			}
 		}
-	}
+
+		/* read tx_power values from fw */
+		u8 *eeprom = dev->mt76.eeprom.data;
+		if (!eeprom[MT_EE_TX0_POWER_2G] || !eeprom[MT_EE_TX0_POWER_5G] || !eeprom[MT_EE_TX0_POWER_6G] ) {
+			use_default = true;
+			goto out;
+		}	
+
+}
 
 out:
 	return mt7996_eeprom_check_or_use_default(dev, use_default);
-- 
2.43.0

